generate.coffee

```diff
@@ -94,14 +94,43 @@ generateAddresses = (total) ->
 generateOrders = (total, customers, addresses, products) ->
   console.log "Generating orders..."
   orders = []
+  orderCounts = [] #a count of orders by customer_id
   for index in [1..total]
-    customer = getRandomItem(customers)
     address = getRandomItem(addresses)
     couponCode = if chance.bool({likelihood: 10}) then getRandomItem(COUPONS) else null
     items = getItems(products, ITEMS_MIN, ITEMS_MAX)
-    createdAt = getRandomDate()
     grandTotal = getCartValue(items)
     shippingAmount = getRandomItem([1.99,3.99,6.99])
+    discountAmount = 0.00
+    if(couponCode)
+      discountAmount = -1*(getRandomItem([0.05,0.1,0.25])*grandTotal).toFixed(2)
```

Comment:
`-1*(getRandomItem([0.05,0.1,0.25])*grandTotal).toFixed(2)` is a lot of logic and should be moved to a sub function.